phoenix-core/src/main/java/org/apache/phoenix/expression/RowValueConstructorExpression.java

```diff
@@ -46,14 +47,30 @@
     private ImmutableBytesWritable literalExprPtr;
     private int partialEvalIndex = -1;
     private int estimatedByteSize;
+    
+    // The boolean field that indicated the object is a literal constant, 
+    // has been repurposed to a bitset and now holds additional information. 
+    // This is to facilitate b/w compat to 4.13 clients.
+    // @see <a href="https://issues.apache.org/jira/browse/PHOENIX-5122">PHOENIX-5122</a> 
+    private BitSet extraFields;
+    
+    private enum ExtraFieldPosition {
```

Comment:
I dislike the ordinal Enum approach as it makes this enum order dependant and for example adding a field to the beginning here could cause backwards compatibility problems as well.  I'd prefer a enum constant with values.

Reply:
I think adding enum constant with values does not help with backward compatibility - since we cannot alter the bit positions anyway. The only benefit of enum with values is for placement in the definition. 
Not sure if there is any value in that, since placement should match with ordinals ideally.

Reply:
I relied on the current unit tests to ascertain that it is functionally the same for > 4.14 client.

Reply:
@dbwong @ChinmaySKulkarni 
Let me know if we need more testing than -

org.apache.phoenix.end2end.InListIT
org.apache.phoenix.end2end.SortOrderIT
org.apache.phoenix.end2end.RowValueConstructorIT